v2/tools/generator/internal/codegen/golden_files_test.go

```diff
@@ -160,12 +160,10 @@ func NewTestCodeGenerator(
 			pipeline.CheckForAnyTypeStageID,
 			pipeline.CreateResourceExtensionsStageID,
 			pipeline.CreateTypesForBackwardCompatibilityID,
-			// TODO: Once the stage is enabled in the pipeline, we may need to remove it here for testing
-			// pipeline.InjectHubFunctionStageID,
-			// pipeline.ImplementConvertibleInterfaceStageId,
 			pipeline.ReportOnTypesAndVersionsStageID,
 			pipeline.ReportResourceVersionsStageID,
-			pipeline.ReportResourceStructureStageId)
+			pipeline.ReportResourceStructureStageId,
+			pipeline.RemoveEmptyObjectsStageId)
```

Comment:
We shouldn't remove stages from the test pipeline unless absolutely necessary - we want those tests to be as close as possible to the real thing. Does including this cause any failures?

Reply:
It did cause failures, but I didn't look that closely into why - I just added it to the ignore. Let me check exactly what caused the failure first and make 100% sure it makes sense.

Reply:
Ah, the issue is that there is a golden test for the `additionalProperties` behavior:

> Here we check that if 'additionalProperties' is explicitly set to 'false' (and there are no other fields) the generated struct will be empty

That expects to generate an empty object. I've updated the test so that it has 2 properties, 1 which is the `additionalProperties: false` object, and the other is a simple string, and re-recorded the test so that we don't have to remove this stage.